tests/model/write_model_test.go

```diff
@@ -0,0 +1,427 @@
+    define parent: [folder, folder#parent] as self
+    define viewer: [user] as self or viewer from parent
+`,
+		code: 2056,
+	},
+	"case21": {
+		model: `
+type user
+type group
+  relations
+    define member: [user] as self
+    define reader as member and allowed
+`,
+		code: 2056,
+	},
+	"case22": {
+		model: `
+type user
+type group
+  relations
+    define member: [user] as self
+    define reader as member or allowed
+`,
+		code: 2056,
+	},
+	"case23": {
+		model: `
+type user
+type group
+  relations
+    define member: [user] as self
+    define reader as allowed but not member
+`,
+		code: 2056,
+	},
+	"case24": {
+		model: `
+type user
+type group
+  relations
+    define member: [user] as self
+    define reader as member but not allowed
+`,
+		code: 2056,
+	},
+	"case25": {
+		model: `
+type user
+type org
+  relations
+    define member as self
+`,
+		code: 2056,
+	},
+	//"case26:": {
+	//	// Parse error
+	//	model: `
+	//type user
+	//type org
+	// relations
+	//   define member: [ ]
+	//`,
+	//},
+	"same_type_fails": {
+		model: `
+type user
+type user
+`,
+		code: 2056,
+	},
+	"difference_includes_itself_in_subtract_fails": {
+		model: `
+type user
+type document
+  relations
+	define viewer: [user] as self but not viewer
+`,
+		code: 2056,
+	},
+	"union_includes_itself_fails": {
+		model: `
+type user
+type document
+  relations
+	define viewer: [user] as self or viewer
+`,
+		code: 2056,
+	},
+	"intersection_includes_itself_fails": {
+		model: `
+type user
+type document
+  relations
+	define viewer: [user] as self and viewer
+`,
+		code: 2056,
+	},
+	"simple_model_succeeds": {
+		model: `
+type user
+type folder
+  relations
+    define viewer: [user] as self
+type document
+  relations
+    define parent: [folder] as self
+    define viewer as viewer from parent
+`,
+	},
+	"no_relations_succeeds": {
+		model: `type user`,
+	},
+	"union_may_contain_repeated_relations": {
+		model: `
+type user
+type document
+  relations
+	define editor: [user] as self
+    define viewer as editor or editor
+`,
+	},
+	"intersection_may_contain_repeated_relations": {
+		model: `
+type user
+type document
+  relations
+	define editor: [user] as self
+    define viewer as editor and editor
+`,
+	},
+	"exclusion_may_contain_repeated_relations": {
+		model: `
+type user
+type document
+  relations
+	define editor: [user] as self
+    define viewer as editor but not editor
+`,
+	},
+}
+
+func TestWriteAuthorizationModel(t *testing.T) {
+	cfg := cmd.MustDefaultConfigWithRandomPorts()
+	cfg.Log.Level = "none"
+	cfg.Datastore.Engine = "memory"
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		if err := cmd.RunServer(ctx, cfg); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	require.NoError(t, err)
+	defer conn.Close()
+
+	// Ensure the service is up before continuing.
+	client := healthv1pb.NewHealthClient(conn)
```

Comment:
We're starting to see this code duplicated a lot in various places in our tests. Maybe it's time to move this into a common helper in `testutils`? WDYT?

Reply:
Done. Thanks!